cli/src/config.rs

```diff
@@ -488,108 +499,148 @@ impl ConfigEnv {
         Ok(())
     }
 
+    fn load_user_config<T: ConfigType>(
+        &mut self,
+        ui: &Ui,
+        dir: &Path,
+    ) -> Result<UserConfig<T>, CommandError> {
+        let user_config = match read_user_config::<T>(dir) {
+            Err(UserConfigError::LegacyRepo) => {
+                write_user_config(dir, &T::default()).map_err(|e| {
+                    internal_error_with_message("Failed to migrate from legacy repo", e)
+                })?;
+                return Ok(UserConfig::Trusted(Default::default()));
+            }
+            other => other,
+        }
+        .map_err(internal_error)?;
+        if let UserConfig::RepoMoved { from, to, .. } = &user_config {
+            let kind = T::kind();
+            writeln!(
+                ui.warning_default(),
+                "The {kind} has moved from {from} to {to}. For security reasons, we have disabled \
+                 the {kind} config.",
+            )?;
+            writeln!(
+                ui.hint_default(),
+                "Run `jj config edit --{kind}` to review and re-enable"
+            )?;
+        } else if matches!(user_config, UserConfig::InvalidSignature(_)) {
+            let kind = T::kind();
+            writeln!(
+                ui.warning_default(),
+                "The {kind} appears to have been created by someone else. For security reasons, \
+                 we have disabled the {kind} config.",
+            )?;
+            writeln!(
+                ui.hint_default(),
+                "Run `jj config edit --{kind}` to review and re-enable"
+            )?;
+        }
+        Ok(user_config)
+    }
+
     /// Sets the directory where repo-specific config file is stored. The path
     /// is usually `.jj/repo`.
-    pub fn reset_repo_path(&mut self, path: &Path) {
+    pub fn reset_repo_path(&mut self, ui: &Ui, path: &Path) -> Result<(), CommandError> {
         self.repo_path = Some(path.to_owned());
-        self.repo_config_path = Some(ConfigPath::new(path.join("config.toml")));
+        // Provide a grace period for which repos without secure user config
+        // information will be automatically migrated to add it.
+        // If we did not provide this grace period, repo configs would be
+        // disabled for every repo created by an older version of jj.
+        // TODO: After the grace period is over, make this act the same
+        // as InvalidSignature or RepoMovedError.
+        let legacy_config = path.join("config.toml");
+        match std::fs::read_to_string(&legacy_config).context(&legacy_config) {
+            Ok(content) => {
+                self.set_repo_config(&content, true)?;
```

Comment:
it would be nice if migration was noted on stdout/err. less confusion if you don't read the release notes carefully.